js/fairdesk.js

```diff
@@ -0,0 +1,1450 @@
+        //       "marginBalanceBtc": "12247.58222118",
+        //       "totalAccountBalance": "439275458.34",
+        //       "totalUnRealizedPnL": "-0.12",
+        //       "accounts": [
+        //         {
+        //           "currency": "USDT",
+        //           "icon": "https://fairdesk-app2public-prod.s3.ap-southeast-1.amazonaws.com/currency/USDT.png",
+        //           "accountBalance": "439275458.34",
+        //           "availBalance": "439275440.23",
+        //           "unRealizedPnL": "-0.12",
+        //           "positionMargin": "10.84",
+        //           "bonus": "495.00",
+        //           "bonusBalance": "0.00",
+        //           "display": "Tether"
+        //         }
+        //       ]
+        //     }
+        //   }
+        return this.parseSwapBalance (response);
+    }
+
+    parseOrderType (type) {
+        const types = {
+            'LIMIT': 'limit',
+            'MARKET': 'market',
+        };
+        return this.safeString (types, type, type);
+    }
+
+    parseTimeInForce (timeInForce) {
+        const timeInForces = {
+            'GTC': 'GTC',
+            'POST_ONLY': 'PO',
+            'IOC': 'IOC',
+            'FOK': 'FOK',
+        };
+        return this.safeString (timeInForces, timeInForce, timeInForce);
+    }
+
+    parseOrders (orders, market = undefined, since = undefined, limit = undefined, params = {}) {
+        const result = [];
+        for (let i = 0; i < orders.length; i++) {
+            const item = orders[i];
+            result.push (this.parseOrder (item));
+        }
+        return result;
+    }
+
+    parseStatus (order) {
+        let value = '';
+        if (this.safeString (order, 'status') === 'FILLED') {
+            value = 'closed';
+        }
+        if (this.safeString (order, 'status') === 'CANCELED') {
+            value = 'canceled';
+        }
+        if (this.safeString (order, 'status') === 'EXPIRED') {
+            value = 'expired';
+        }
+        if (this.safeString (order, 'status') === 'NEW') {
+            value = 'open';
+        }
+        return value;
+    }
+
+    parseOrder (order, market = undefined) {
+        const symbol = this.safeString (order, 'symbol');
+        const status = this.parseStatus (order); // 'open', 'closed', 'canceled', 'expired', 'rejected'
+        const side = this.safeStringLower (order, 'side');
+        const positionSide = this.safeStringLower (order, 'positionSide');
+        const type = this.parseOrderType (this.safeString (order, 'type'));
+        const price = this.parseNumber (this.safeString (order, 'price') === '--' ? '' : this.safeString (order, 'price'));
+        const amount = this.safeNumber (order, 'origQty');
+        const remaining = amount - this.safeNumber (order, 'executedQty');
+        const timestamp = this.safeInteger (order, 'transactTime');
+        const timeInForce = this.parseTimeInForce (this.safeString (order, 'timeInForce'));
+        return {
```

Comment:
Let's use `safeOrder` here as we do in the other `parseOrder` implementations with strings..